credentialdigger/client.py

```diff
@@ -883,11 +1042,15 @@ def _scan(self, repo_url, scanner, models=None, force=False, debug=False,
                     curr_d['file_name'], curr_d['commit_id'],
                     curr_d['line_number'], curr_d['snippet'], repo_url,
                     curr_d['rule_id'], curr_d['state'])
+                if similarity:
+                    self.add_embedding(new_id, repo_url)
```

Comment:
this is just for the debug mode, where the if part is made of multiple calls to add_discovery, so I made it similar, this way one can precisely locate where a bug might arise. I don't think there's a significant difference in speed between add_embedding vs add_embeddings, especially since the cursor is not closed